tasks/authorize.yml

```diff
@@ -4,9 +4,10 @@
     user: "postgres"
     key: "{{ backup_server_pub_key['content']|b64decode }}"
     state: present
-  delegate_to: "{{ hostvars[item].ansible_host | default(item) }}"
-  with_items: "{{ groups[barman_database.ansible_group] }}"
+  delegate_to: "{{ item }}"
+  with_items: "{{ groups[barman_database.ansible_group|default('')] | default(barman_database.authorize_ssh_hosts) }}"
```

Comment:
If `barman_database.ansible_group` is not defined, it produces `groups[''] | default(barman_database.authorize_ssh_hosts)`

Have you tested it works as expected and does not raise an error because `''` group does not exist?

Reply:
I have tested yes. it works in both case `ansible_group` defined or `ansible_group` not defined